Tools/cases_generator/analyzer.py

```diff
@@ -1079,6 +1081,36 @@ def add_instruction(name: str) -> None:
     return instmap, len(no_arg), min_instrumented
 
 
+def is_instruction_size_macro(token: lexer.Token) -> bool:
+    return token.kind == "IDENTIFIER" and token.text == "INSTRUCTION_SIZE"
+
+
+def get_instruction_size_for_uop(instructions: dict[str, Instruction], uop: Uop) -> int | None:
+    """Return the size of the instruction that contains the given uop or
+    `None` if the uop does not contains the `INSTRUCTION_SIZE` macro.
+
+    If there is more than one instruction that contains the uop,
+    ensure that they all have the same size.
+    """
+    if not (token := next((t for t in uop.body if is_instruction_size_macro(t)), None)):
+        return None
+
+    size = None
+    for inst in instructions.values():
+        if uop in inst.parts:
+            if size is None:
+                size = inst.size
+            if size != inst.size:
```

Comment:
```suggestion
            elif size != inst.size:
```

Reply:
I changed this to an `if` because otherwise mypy complains that the `elif` is unreachable:

```
Tools/cases_generator/analyzer.py:1103: error: Statement is unreachable  [unreachable]
```

Reply:
OK, leave it then.